vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -247,6 +260,63 @@ public void addCategory(IVSImage image, ImageCategory category) {
     public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
+    }   
+    
+    /**
+     * Method to store an image in the file location
+     * 
+     *@param image - The image data as a byte array
+     *@param filename - The name of the file to be stored
+     *@return {@link IVSImage} instance
+     */
+    @Override
+    public IVSImage storeImage(byte[] image, String filename) {
+
+        IVSImage storedImage = null;
+        if (image != null && image.length > 0) {
+            Tika tika = new Tika();
+            String contentType = tika.detect(image);
+            storedImage = imageFactory.createImage(filename, contentType);
+            storedImage = imageRepo.save((VSImage) storedImage);
+        }
+
+        CreationReturnValue returnValue = new CreationReturnValue();
+        returnValue.setErrorMsgs(new ArrayList<>());
+
+        if (storedImage != null) {
+            String relativePath = null;
+            try {
+                relativePath = storage.storeFile(image, filename, storedImage.getId());
+            } catch (FileStorageException e) {
+                logger.error(DEFAULT_IMAGE_EXCEPTION,e);
+                returnValue.getErrorMsgs().add(DEFAULT_IMAGE_EXCEPTION + e.getMessage());
```

Comment:
if this is the case and the image can't be stored, the vsimage object should not be stored either.

Reply:
?

Reply:
in line 291 deleting the vsimage object if an exception occurs, as we are using its id to store the image